packages/indexer/src/lib/document-processor.ts

```diff
@@ -47,8 +24,11 @@ export class DocumentProcessor {
     if (type === 'text/plain' || type === 'text/markdown') {
```

Comment:
I wonder if it makes sense to have something like a registerDocumentHandler(type, extract: () => Promise<ContentPage[]>) to enable a more pluggable handling?

Reply:
I like this idea, in terms of extensibility.

Reply:
Good idea, I'll add this

Reply:
Done